### PR TITLE
MacOSRequirement: add to_json method

### PR DESCRIPTION
**Pull request by EricFromCanada**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/master/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/master/Library/Homebrew/test/PATH_spec.rb).
- [x] Have you successfully run `brew style` with your changes locally?
- [x] Have you successfully run `brew tests` with your changes locally?

-----

This adds a customized JSON representation for `MacOSRequirement` that's been needed for casks since #6366, as seen in [this automated commit](https://github.com/Homebrew/formulae.brew.sh/commit/c00aea3769b7ce6541da2486c3679eb6e1832380#diff-85022ccfab41a971058dcbcadb3ffee8) to Homebrew/formulae.brew.sh (and each subsequent commit). 

Before:
```
$ brew cask info --json=v1 1password | jq ".[0].depends_on"
{
  "macos": "#<MacOSRequirement:0x00007f960e8b33b8>"
}
$ brew cask info --json=v1 openzfs | jq ".[0].depends_on"
{
  "macos": "#<MacOSRequirement:0x00007f8eec420d38>"
}
```
After:
```
$ brew cask info --json=v1 1password | jq ".[0].depends_on"
{
  "macos": {
    ">=": [
      "10.12"
    ]
  }
}
$ brew cask info --json=v1 openzfs | jq ".[0].depends_on"
{
  "macos": {
    "==": [
      "10.9",
      "10.10",
      "10.11",
      "10.12",
      "10.13",
      "10.14"
    ]
  }
}
```

### CODE REVIEW

File: Library/Homebrew/requirements/macos_requirement.rb [hunk 1]
```diff
@@ -59,4 +59,10 @@ def display_s
 
     "macOS #{@comparator} #{@version}"
   end
+
+  def to_json(*args)
+    {
+      @comparator.to_s => (@version.is_a?(Array) ? @version.map(&:to_s) : [@version.to_s]),
```
Comment (MikeMcQuaid) [inline Library/Homebrew/requirements/macos_requirement.rb]: Can you split this into an `if`/`else` and use some intermediate variables? It's pretty hard to follow as-is. Thanks!

Comment (EricFromCanada) [reply Library/Homebrew/requirements/macos_requirement.rb]: Have a look. ↓
